src/plugins/chart_expressions/expression_gauge/public/components/utils/use_gauge_size_by_type.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect } from 'react';
+import { ChartSizeSpec } from '@kbn/chart-expressions-common';
+import { GaugeShape, GaugeShapes } from '../../../common';
+
+export const useGaugeSizeByType = (type: GaugeShape, setChartSize: (d: ChartSizeSpec) => void) => {
+  useEffect(() => {
+    const maxDimensions: Record<GaugeShape, ChartSizeSpec> = {
+      [GaugeShapes.ARC]: {
+        maxDimensions: {
+          x: { value: 600, unit: 'pixels' },
+          y: { value: 600, unit: 'pixels' },
+        },
+        aspectRatio: {
+          x: 2,
+          y: 1.25,
+        },
+      },
+      [GaugeShapes.MAJOR_ARC]: {
+        maxDimensions: {
+          x: { value: 600, unit: 'pixels' },
+          y: { value: 600, unit: 'pixels' },
+        },
+        aspectRatio: {
+          x: 1.1,
+          y: 1,
+        },
+      },
+      [GaugeShapes.CIRCLE]: {
+        maxDimensions: {
+          x: { value: 600, unit: 'pixels' },
+          y: { value: 600, unit: 'pixels' },
+        },
+      },
```

Comment:
In testing, I noticed the circle gauge didn't appear to have an aspect ratio applied. This can cause the visualization to appear oddly in its container at smaller viewport widths. Would it be possible to add a 1:1 ratio to ensure it retains its square shape?

![CleanShot 2024-03-19 at 15 29 57](https://github.com/elastic/kibana/assets/3884767/9cbb9865-655b-4cb0-a845-4b4a4e5179f1)

Reply:
Fixed in f53b2477834c67cbba46adab5a72744c6ceeb5c2

![Zight Recording 2024-03-21 at 01 47 08 PM](https://github.com/elastic/kibana/assets/19007109/4656b535-ca24-49d2-be8d-054998b9bb3d)